### ISSUE TITLE
Scout config generation bug

### ISSUE THREAD

**Issue by dnil** [Clinical-Genomics/cg#1491]
## Description

Scout MIP uploads appear broken atm. 

From @Karl-Svard:
```
Could not find any files matching the tags
Created new bundle version dir: /home/proj/production/housekeeper-bundles/usefuloyster/2022-05-18
Linked file: /home/proj/production/rare-disease/cases/usefuloyster/scout_load.yaml -> /home/proj/production/housekeeper-bundles/usefuloyster/2022-05-18/scout_load.yaml
Added scout load config to housekeeper: /home/proj/production/rare-disease/cases/usefuloyster/scout_load.yaml
----------------- UPLOAD -----------------------
Fetch latest version from bundle usefuloyster
uploading case usefuloyster to scout
Traceback (most recent call last):
  File "/home/proj/bin/conda/envs/P_main/bin/cg", line 8, in <module>
    sys.exit(base())
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/cg/cli/upload/scout.py", line 47, in scout
    context.invoke(upload_case_to_scout, case_id=case_id, re_upload=re_upload)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/cg/cli/upload/scout.py", line 140, in upload_case_to_scout
    scout_api.upload(scout_load_config=scout_config_file.full_path, force=re_upload)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/cg/apps/scout/scoutapi.py", line 32, in upload
    data = yaml.safe_load(stream)
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/constructor.py", line 51, in get_single_data
    return self.construct_document(node)
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/constructor.py", line 60, in construct_document
    for dummy in generator:
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/constructor.py", line 413, in construct_yaml_map
    value = self.construct_mapping(node)
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/constructor.py", line 218, in construct_mapping
    return super().construct_mapping(node, deep=deep)
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/constructor.py", line 143, in construct_mapping
    value = self.construct_object(value_node, deep=deep)
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/home/hiseq.clinical/.local/lib/python3.7/site-packages/yaml/constructor.py", line 429, in construct_undefined
    node.start_mark)
yaml.constructor.ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/object/apply:cg.constants.subject.PhenotypeStatus'
  in "/home/proj/production/housekeeper-bundles/usefuloyster/2022-05-18/scout_load.yaml", line 27, column 14
```

This appears to derive from some odd type tags being exported into the scout load config upon generation in CG (@Vince-janv):
![Screenshot 2022-05-20 at 17 04 40](https://user-images.githubusercontent.com/758570/169639707-155f33dd-3460-4d26-ae86-d3323831a7ab.png)

It appears those values recently changed as a commendable refactor-stuff-while-fixing-a-real-issue in https://github.com/Clinical-Genomics/cg/pull/1464, and that inadvertently at some intermediate stage `PlinkGender` is confused with a `Gender` and a `PlinkPhenotypeStatus` with a `PhenotypeStatus`, which was likely handled by implicit conversion via Int or Str before.

## Suggested solution

Either revert or fix the Pydantic cast.

## This can be closed when

MIP cases can be uploaded to scout using CG again.

**Comment by dnil** [comment]
Heh, I'm too slow: this appears to fix the issue https://github.com/Clinical-Genomics/cg/pull/1490!

**Comment by dnil** [comment]
Perhaps PyYaml 5.1 didn't `repr()` them as `5.3.1` did? Anyhow, fixed by #1492!
